xds/src/main/java/io/grpc/xds/XdsClusterResource.java

```diff
@@ -177,8 +180,7 @@ static CdsUpdate processCluster(Cluster cluster,
         ImmutableMap.copyOf(cluster.getMetadata().getFilterMetadataMap()));
 
     try {
-      ImmutableMap<String, Object> parsedFilterMetadata =
-          parseClusterMetadata(cluster.getMetadata());
+      ImmutableMap<String, Object> parsedFilterMetadata = parseMetadata(cluster.getMetadata());
       updateBuilder.parsedMetadata(parsedFilterMetadata);
     } catch (InvalidProtocolBufferException e) {
```

Comment:
Catch `ResourceInvalidException` as well, to provide a better error message. Ditto elsewhere